packages/core-data/src/selectors.js

```diff
@@ -104,7 +104,20 @@ export function getEntity( state, kind, name ) {
  * @return {Object?} Record.
  */
 export function getEntityRecord( state, kind, name, key ) {
-	return get( state.entities.data, [ kind, name, 'queriedData', 'items', key ] );
+	const record = get( state.entities.data, [
+		kind,
+		name,
+		'queriedData',
+		'items',
+		key,
+	] );
+	return (
+		record &&
+		Object.keys( record ).reduce( ( acc, _key ) => {
+			acc[ _key ] = get( record[ _key ], 'raw', record[ _key ] );
+			return acc;
+		}, {} )
+	);
```

Comment:
This is an interesting one, I believe you're trying to always return the "raw" value for these properties. Makes me wonder if it should be the responsibility of the caller to do this too. 

What if I want the "rendered" value, what If there's a custom property with "raw" having a different meaning.

Reply:
I guess the difficulty is that the "edits" won't align with the persisted property in that case.
I wonder if we should "define" these properties in the entity config.

Reply:
Yeah, we can do something like:

>"define" these properties in the entity config.

If this ever becomes a problem. Or just make another selector.

Reply:
Can we clarify how this work, do we always use the "raw" value. Return raw values from selectors, edit raw values...? Maybe we can document the behavior somewhere (comment or something). Because it's really unclear if you're not familiar with the problem.

Reply:
We edit and select raw values, so yeah we always use the raw values.

Reply:
I added comments everywhere it happens to make it clearer:

https://github.com/WordPress/gutenberg/pull/16932/commits/960f44d044313c95832b3409e43ff62a40c77418

Reply:
Are we exposing the entities "rendered" values anywhere?

- Also is the change to `getEntityRecord` a breaking change? What if a third party plugin was using the "rendered" value from that call?
- Why `getEntityRecords` returns a different format than `getEntityRecord`?

Honestly, it makes me wonder we shouldn't touch the returned value. The idea being these selectors return the persisted values unchanged. Is there ways we could do the "merging" only once we need the "edited" values?

Reply:
>Are we exposing the entities "rendered" values anywhere?
>Why getEntityRecords returns a different format than getEntityRecord?

"Latest Posts" uses `getEntityRecords` and the returned posts' `title.rendered`s.

>Also is the change to getEntityRecord a breaking change? What if a third party plugin was using the "rendered" value from that call?

I doubt it, but it's a valid concern. I would rather hide this `.raw`/`.rendered` thing from Gutenberg, it's really awkward because edits are not in that format and it's not clear what determines what `.rendered` contains. But, if it's a big BC issue, we can make a new `getRawEntityRecord` selector.

>Is there ways we could do the "merging" only once we need the "edited" values?

Yeah, we can just have a different selector, `getRawEntityRecord`. What do you think?

Reply:
> Yeah, we can just have a different selector, getRawEntityRecord. What do you think?

This feels safer for me. but I'm wondering where do we need it? Can't we use `getEditedEntityRecord` in these situations?

Reply:
Everywhere we compare or merge with edits.

Reply:
> Everywhere we compare or merge with edits.

This places are internal to the core module right? I mean I wouldn't mind a new `getRawEntityRecord` but if we can avoid this being a public API. It's best.

Reply:
It doesn't have to be public. It's just to avoid rewriting it everywhere and be able to search for where it happens easily.

Reply:
https://github.com/WordPress/gutenberg/pull/16932/commits/0ca7b0fd49781d3f911857bc2420f26813b4b941

I think it will be common for other packages to want to get the raw values, so I made it public if that's OK?

Reply:
👍